File: file_handler.py
from pathlib import Path

# Contents of this file were written with the help of ChatGPT

def get_data_fname(header_fname):
    path = Path(header_fname)
    return str(path.with_name(path.stem + "Data.dat"))


def remove_path(fname):
    path = Path(fname)
    return path.name


def add_path(src_fname, dst_fname):
    src_path = Path(src_fname)
    new_path = src_path.parent / dst_fname
    return str(new_path)


def remove_step_counter(fname):
    path = Path(fname)
    stem = path.stem
    i = len(stem) - 1
    while i >= 0 and stem[i].isdigit():
        i -= 1
    new_stem = stem[:i+1]
    return str(path.with_stem(new_stem))


def create_dir_for_file(fname):
    path = Path(fname)
    path.parent.mkdir(parents=True, exist_ok=True)


def get_last_header_fname(base_name, dir="."):
    last_num = 0
    last_file = base_name + str(last_num)
    for path in Path(dir).glob(f"{base_name}*"):
        suffix = path.stem[len(base_name):]
        if suffix.isdigit():
            num = int(suffix)
            if num > last_num:
                last_num = num
                last_file = path
    return last_file

File: FileHandler.cpp
#include "FileHandler.H"

#include <filesystem>

// Contents of this file were written with the help of ChatGPT

std::string getDataFilename(const std::string& headerFilename)
{
    std::filesystem::path path(headerFilename);
    path.replace_filename(path.stem().string() + "Data.dat");
    return path.string();
}

std::string removePath(const std::string& filename)
{
    std::filesystem::path path(filename);
    return path.filename().string();
}

std::string addPath(const std::string& srcFilename, const std::string& dstFilename)
{
    std::filesystem::path srcPath(srcFilename);
    std::filesystem::path newPath = srcPath.parent_path() / dstFilename;
    return newPath.string();
}

std::string addStepCounter(const std::string& filename, const int step)
{
    std::filesystem::path path(filename);
    std::string dir = path.parent_path().string();
    std::string stem = path.stem().string();
    std::string extension = path.extension().string();
    return dir + "/" + stem + std::to_string(step) + extension;
}

void createDirForFile(const std::string& filename)
{
    std::filesystem::path path(filename);
    std::filesystem::path dir = path.parent_path();
    if(!std::filesystem::exists(dir))
    {
        std::filesystem::create_directory(dir);
    }
}
